### PR TITLE
kv, store: remove `Rollback` in `RunInNewTxn`

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve? 
 In the RunInNewTxn function, the error of `invalid transaction` will be reported when the commit fails.
```
2018/11/07 11:37:41.755 txn.go:65: [warning] [kv] Retry txn 404118742386346212 original txn 404118742386346212 err [try again later]: WriteConflict: startTS=404118742386346212, conflictTS=404118742399452259, key=[]byte{0x6d, 0x44, 0x42, 0x3a, 0x31, 0x31, 0x31, 0x0, 0x0, 0xfd, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x68, 0x54, 0x49, 0x44, 0x3a, 0x31, 0x31, 0x33, 0x0, 0xfe} primary=[]byte{0x6d, 0x44, 0x42, 0x3a, 0x31, 0x31, 0x31, 0x0, 0x0, 0xfd, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x68, 0x54, 0x49, 0x44, 0x3a, 0x31, 0x31, 0x33, 0x0, 0xfe}
2018/11/07 11:37:41.755 terror.go:342: [error] [kv:8]invalid transaction
github.com/pingcap/errors.AddStack
    /home/ubuntu/projects/go/pkg/mod/github.com/pingcap/errors@v0.11.0/errors.go:174
github.com/pingcap/errors.Trace
    /home/ubuntu/projects/go/pkg/mod/github.com/pingcap/errors@v0.11.0/juju_adaptor.go:12
github.com/pingcap/tidb/kv.RunInNewTxn
    /home/ubuntu/projects/go/src/github.com/pingcap/tidb/kv/txn.go:67
...
```

### What is changed and how it works?
If "Commit" is completed, it will call txn's `close`. So we can remove `Rollback` and the print log in `RunInNewTxn`.

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: kv/txn.go [hunk 1]
```diff
@@ -48,8 +47,6 @@ func RunInNewTxn(store Storage, retryable bool, f func(txn Transaction) error) e
 
 		err = f(txn)
 		if err != nil {
-			err1 := txn.Rollback()
```
Comment (jackysp) [inline kv/txn.go]: Is it safe to remove this?

Comment (crazycs520) [reply kv/txn.go]: I have the same question.

File: store/tikv/2pc_fail_test.go [hunk 2]
```diff
@@ -33,6 +34,10 @@ func (s *testCommitterSuite) TestFailCommitPrimaryRpcErrors(c *C) {
 	err = t1.Commit(context.Background())
 	c.Assert(err, NotNil)
 	c.Assert(terror.ErrorEqual(err, terror.ErrResultUndetermined), IsTrue, Commentf("%s", errors.ErrorStack(err)))
+
+	// We don't need to call "Rollback" after "Commit" fails.
```
Comment (shenli) [inline store/tikv/2pc_fail_test.go]: The comment confuses me. Why do you call the `Rollback()` in line 39?

File: kv/txn.go (hunk 1 above)
Comment (winkyao) [reply kv/txn.go]: I don't think this change is appropriate. We should keep the `txn.Rollback()` and just check if the error is kv.ErrInvalidTxn is better.

File: kv/txn.go [hunk 3]
```diff
@@ -63,8 +60,6 @@ func RunInNewTxn(store Storage, retryable bool, f func(txn Transaction) error) e
 		}
 		if retryable && IsRetryableError(err) {
 			log.Warnf("[kv] Retry txn %v original txn %v err %v", txn, originalTxnTS, err)
-			err1 := txn.Rollback()
```
Comment (winkyao) [inline kv/txn.go]: ditto

File: store/tikv/2pc_fail_test.go (hunk 2 above)
Comment (zimulala) [reply store/tikv/2pc_fail_test.go]: This test is used to prove why the rollback was removed in "RunInNewTxn".

File: kv/txn.go (hunk 3 above)
Comment (zimulala) [reply kv/txn.go]: I said

> If "Commit" is completed, it will call txn's `close`. So we can remove Rollback and the print log in RunInNewTxn.

**Comment by zimulala** [comment]
PTAL @jackysp @crazycs520

**Pull request merged**